src/gt4py/frontend/gtscript_frontend.py

```diff
@@ -1652,12 +1657,31 @@ def annotate_definition(definition):
                 api_annotations.append(dtype_annotation)
 
         nonlocal_symbols, imported_symbols = GTScriptParser.collect_external_symbols(definition)
-        canonical_ast = gt_meta.ast_dump(definition)
+        ast_func_def = gt_meta.get_ast(definition).body[0]
+        canonical_ast = gt_meta.ast_dump(ast_func_def)
+
+        # Gather temporary
+        temp_annotations = {}
+        temp_init_values = {}
+        ann_assigns = tuple(filter(lambda stmt: isinstance(stmt, ast.AnnAssign), ast_func_def.body))
+        for ann_assign in ann_assigns:
+            assert isinstance(ann_assign.target, ast.Name)
+            name = ann_assign.target.id
+
+            source = gt_meta.ast_unparse(ann_assign.annotation)
+            descriptor = eval(source, {"Field": gtscript.Field, "IJK": gtscript.IJK, "np": np})
```

Comment:
Well this is aweful, but since the context will not contain any variables referenced inside annotations (as inspect.getclosurevars and alike will ignore annotations) this is reasonable. Please also add the other axes, i.e. `"I", "J", "K", "IJ", "IK", "JK"` and a todo. I think the proper (but arguably also not great) way to handle this is to use `symtable.symtable` on all annotations and grabbing the referenced variables from the stack frame in which the stencil is defined (at the location where the closure vars are retrieved).